docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+
++--------------------------------------------------------------------------------------------------+
+ NO |  MODULE CODE   |                                 MODULE TITLE                       
++--------------------------------------------------------------------------------------------------+
+ 1  |     CS1231     |                             Discrete Structures                    
+ 2  |     CS2100     |                            Computer Organisation                   
+ 3  |     CS2113     |              Software Engineering & Object-Oriented Programming    
+ 4  |     CS3235     |                              Computer Security                     
++--------------------------------------------------------------------------------------------------+
+Total modules: 4
++--------------------------------------------------------------------------------------------------+
+
+root :
+lsc -m cs3235
+Here are what you are looking for...
+
++--------------------------------------------------------------------------------------------------+
+ NO |     MODULE     |                                CATEGORY                                | PTY
++--------------------------------------------------------------------------------------------------+
+ 1  |     CS3235     |                               Assignment                               |  4
+ 2  |     CS3235     |                                  Lab                                   |  3
+ 3  |     CS3235     |                                Lecture                                 |  1
+ 4  |     CS3235     |                                Tutorial                                |  2
++--------------------------------------------------------------------------------------------------+
+Total categories: 5
++--------------------------------------------------------------------------------------------------+
+```
+
+1. James has two choices:
+
+   1. enter `cd cs3235` to enter the module directory then enter `addc misc` to add the category.
+   2. enter `addc misc -m cs3235` to add the category at the root directory.
+
+   Suppose James use the first method, after the second input, the input is parsed as an **add task** command and executed, the `AddCategoryCommand#execute()` will call ``AddCategoryCommand#getParentDirectory()` to get the current parent directory, then it will instantiate an `Category` object with the name "misc". After which `CategoryManager#add()` will be called to add the new object. In the `CategoryManager#add()` method, it will call `CategoryManager#contains()` method to check if the current parent directory contains the category with name "misc", finally add the object into the `ArrayList` of `categoryList`.
+
+2. James receive the following feedback:
+
+   ```
+   root :
+   addc misc
+   SUCCESS!! Category misc is created.
+   ```
+
+
+
+Below is a *sequence diagram* to illustrate the above example scenario.  
+
+```
+to-do: add the sequence diagram
```

Comment:
Yes please add more sequence diagram to aid the reader understand the feature more 👍